File: backend/app/models/product.py
from pydantic import BaseModel, Field
from typing import Optional, List
from datetime import datetime


class ProductBase(BaseModel):
    market_id: str
    name: str = Field(..., max_length=200)
    description: Optional[str] = None
    price: float = Field(..., gt=0)
    stock_quantity: int = Field(..., ge=0)
    category: str
    sub_category: Optional[str] = None
    tags: List[str] = []
    images: List[str] = []


class ProductCreate(ProductBase):
    pass


class ProductInDB(ProductBase):
    id: str
    created_at: datetime
    updated_at: datetime


class ProductPublic(ProductBase):
    id: str
    created_at: datetime
    updated_at: datetime


class ProductUpdate(BaseModel):
    name: Optional[str] = Field(None, max_length=200)
    description: Optional[str] = None
    price: Optional[float] = Field(None, gt=0)
    stock_quantity: Optional[int] = Field(None, ge=0)
    category: Optional[str] = None
    sub_category: Optional[str] = None
    tags: Optional[List[str]] = None
    images: Optional[List[str]] = None 

File: frontend/src/types/product.ts
export interface ProductPublic {
  id: string;
  market_id: string;
  name: string;
  description?: string;
  price: number;
  stock_quantity: number;
  category: string;
  sub_category?: string;
  tags: string[];
  images: string[];
  created_at: string;
  updated_at: string;
}

export interface ProductCreate {
  market_id: string;
  name: string;
  description?: string;
  price: number;
  stock_quantity: number;
  category: string;
  sub_category?: string;
  tags?: string[];
  images?: string[];
}

export interface ProductUpdate {
  name?: string;
  description?: string;
  price?: number;
  stock_quantity?: number;
  category?: string;
  sub_category?: string;
  tags?: string[];
  images?: string[];
}

export interface ProductFilters {
  category?: string;
  sub_category?: string;
  tags?: string[];
  min_price?: number;
  max_price?: number;
  in_stock?: boolean;
  search?: string;
} 